cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -16,54 +16,56 @@
  * event and wait for all constraints to be satisfied to launch the Workflow. Set the schedule's trigger with
  * methods {@link #triggerByTime(String)} or {@link #triggerOnPartitions(String, int)}.
  * Once the schedule's trigger is set, the schedule will be added to the app with all of the information
- * set in this configurer. To set the constraints for the schedule,
- * use methods {@link #limitConcurrentRuns(int)}, {@link #delayRun(long)}, {@link #setTimeRange(String, String)},
- * {@link #setTimeRange(String, String, TimeZone)}, and {@link #setDurationSinceLastRun(long)}.
+ * set in this builder. To set the constraints for the schedule,
+ * use methods {@link #withConcurrency(int)}, {@link #withDelay(long)}, {@link #withTimeWindow(String, String)},
+ * {@link #withTimeWindow(String, String, TimeZone)}, and {@link #withDurationSinceLastRun(long)}.
  * If no constraint is specified, the schedule will immediately launch the Workflow once the schedule is triggered.
  */
-public interface ScheduleConfigurer {
+public interface ScheduleBuilder {
 
   /**
    * Set the description of the schedule.
    *
    * @param description given description of the schedule
-   * @return {@link ScheduleConfigurer} containing the given description
+   * @return {@link ScheduleBuilder} containing the given description
    */
-  ScheduleConfigurer setDescription(String description);
+  ScheduleBuilder setDescription(String description);
 
   /**
    * Set properties of the schedule.
    *
    * @param properties a {@link Map} with property names of class {@link String} as keys and
    *                   property values of class {@link String} as values
-   * @return {@link ScheduleConfigurer} containing the given properties
+   * @return {@link ScheduleBuilder} containing the given properties
    */
-  ScheduleConfigurer setProperties(Map<String, String> properties);
+  ScheduleBuilder setProperties(Map<String, String> properties);
 
   /**
    * Set the max number of concurrently running {@link co.cask.cdap.api.workflow.Workflow}
    * can be launched by this schedule.
    *
    * @param max the max number of concurrently running Workflows allowed to be launched by this schedule
-   * @return {@link ScheduleConfigurer} containing the given max number of concurrent runs
+   * @return {@link ConstraintProgramScheduleBuilder} containing the given max number of concurrent runs
    */
-  ScheduleConfigurer limitConcurrentRuns(int max);
+  ConstraintProgramScheduleBuilder withConcurrency(int max);
 
   /**
    * Set a certain amount of delay passed after the schedule is triggered, before launching the Workflow.
    *
    * @param delayMillis delay in milliseconds to wait after the schedule is triggered before launching the Workflow
-   * @return {@link ScheduleConfigurer} containing the given delay
+   * @return {@link ConstraintProgramScheduleBuilder} containing the given delay
```

Comment:
should be `ScheduleBuilder`. Also better to explain why `ScheduleBuilder` is returned so that user can't use setWaitUntilMet and abortIfNotMet